Update changelog for 1.9.0-rc1 (#13067)
hashicorp/vault: CHANGELOG.md

* **KV patch (Tech Preview)**: Add partial update support for the `/<mount>/data/:path` kv-v2
  endpoint through HTTP `PATCH`.  A new `patch` ACL capability has been added and
  is required to make such requests. [[GH-12687](https://github.com/hashicorp/vault/pull/12687)]
* **Key Management Secrets Engine (Enterprise)**: Adds support for distributing and managing keys in GCP Cloud KMS. [[GH-2158](https://github.com/hashicorp/vault/pull/2158)]
* **Namespace Locking (Enterprise)**: Adds support for preventing requests against particular namespaces. [[GH-2213](https://github.com/hashicorp/vault/pull/2213)]
* **OIDC Identity Provider (Tech Preview)**: Adds support for Vault to be an OpenID Connect (OIDC) provider. [[GH-12932](https://github.com/hashicorp/vault/pull/12932)]
* **Oracle Database UI**: The UI now supports adding and editing Oracle connections in the database secret engine [[GH-12752](https://github.com/hashicorp/vault/pull/12752)]
* **Postgres Database UI**: The UI now supports adding and editing Postgres connections in the database secret engine [[GH-12945](https://github.com/hashicorp/vault/pull/12945)]

> * **Deduplicate Token With Entities in Activity Log**: Vault tokens without entities are now tracked with client IDs and deduplicated in the Activity Log [[GH-12820](https://github.com/hashicorp/vault/pull/12820)]

Reviewer: ```suggestion
* **Deduplicate Token Without Entities in Activity Log**: Vault tokens without entities are now tracked with client IDs and deduplicated in the Activity Log [[GH-12820](https://github.com/hashicorp/vault/pull/12820)]
```

## 1.9.0-rc1
### November 5, 2021

CHANGES:

* expiration: VAULT_16_REVOKE_PERMITPOOL environment variable has been removed. [[GH-12888](https://github.com/hashicorp/vault/pull/12888)]
* expiration: VAULT_LEASE_USE_LEGACY_REVOCATION_STRATEGY environment variable has
  been removed. [[GH-12888](https://github.com/hashicorp/vault/pull/12888)]
* go: Update go version to 1.17.2
* secrets/ssh: Roles with empty allowed_extensions will now forbid end-users
  specifying extensions when requesting ssh key signing. Update roles setting
  allowed_extensions to '*' to permit any extension to be specified by an end-user. [[GH-12847](https://github.com/hashicorp/vault/pull/12847)]

FEATURES:

* **Customizable HTTP Headers**: Add support to define custom HTTP headers for root path (`/`) and also on API endpoints (`/v1/*`) [[GH-12485](https://github.com/hashicorp/vault/pull/12485)]

> * **Elasticsearch in the UI**: Elasticsearch DB is now supported by the UI [[GH-12672](https://github.com/hashicorp/vault/pull/12672)]

Reviewer 2: Can we make it more clear this is inside the database secrets engine?

Author: Any suggestions on structuring the message? There are a few that are like this.

Author: Updated in https://github.com/hashicorp/vault/pull/13067/commits/9cdb4f9f01da0fb623ae845f3295964eb5768e78. WDYT?

* **KV Custom Metadata**: Add ability in kv-v2 to specify version-agnostic custom key metadata via the
  metadata endpoint. The data will be present in responses made to the data endpoint independent of the
  calling token's `read` access to the metadata endpoint. [[GH-12907](https://github.com/hashicorp/vault/pull/12907)]

> * **KV patch**: Add partial update support the for the `/<mount>/data/:path` kv-v2

Reviewer: ```suggestion
* **KV patch**: Add partial update support for the `/<mount>/data/:path` kv-v2
```

  endpoint through HTTP `PATCH`.  A new `patch` ACL capability has been added and

>   is required to make such requests. [[GH-12687](https://github.com/hashicorp/vault/pull/12687)]

Reviewer 2: Should we call out here that this is in tech preview status?

Author: Updated

* **Key Management Secrets Engine (Enterprise)**: Adds support for distributing and managing keys in GCP Cloud KMS. [[GH-2158](https://github.com/hashicorp/vault/pull/2158)]

> * **Namespaces (Enterprise)**: Adds support for locking Vault API for particular namespaces. [[GH-2213](https://github.com/hashicorp/vault/pull/2213)]

Reviewer 2: ```suggestion
* **Namespaces (Enterprise)**: Adds support for locking the Vault API for particular namespaces. [[GH-2213](https://github.com/hashicorp/vault/pull/2213)]
```

> * **Namespaces (Enterprise)**: Adds support for locking Vault API for particular namespaces. [[GH-2213](https://github.com/hashicorp/vault/pull/2213)]

Reviewer: ```suggestion
* **Namespace Locking (Enterprise)**: Adds support for preventing requests against particular namespaces. [[GH-2213](https://github.com/hashicorp/vault/pull/2213)]
```

Author: Updated

> * **OIDC Authorization Code Flow**: The Vault UI now supports OIDC Authorization Code Flow [[GH-12800](https://github.com/hashicorp/vault/pull/12800)]

Reviewer 2: Is this the same feature as the one below?

Author: Yeah I'll remove this one and keep the other

* **OIDC Identity Provider (Tech Preview)**: Adds support for Vault to be an OpenID Connect (OIDC) provider. [[GH-12932](https://github.com/hashicorp/vault/pull/12932)]

> * **Oracle DB in the UI**: Oracle DB connection is now supported in the UI [[GH-12752](https://github.com/hashicorp/vault/pull/12752)]

Reviewer 2: Same comment here as with elastic search and postgres below

* **Postgres in the UI**: Postgres DB is now supported by the UI [[GH-12945](https://github.com/hashicorp/vault/pull/12945)]
* **Transform (Enterprise)**: Add advanced features for encoding and decoding for Transform FPE

> * **Transform (Enterprise)**: Add a `reference` field to batch items, and propogate it to the response

Reviewer 2: Feels like an improvement vs a feature

Author: Moved

> * Add ClientID to Token With Entities in Activity Log: Vault tokens without entities are now tracked with client IDs and deduplicated in the Activity Log [[GH-12820](https://github.com/hashicorp/vault/pull/12820)]

Reviewer: ```suggestion
* Deduplicate Token With Entities in Activity Log: Vault tokens without entities are now tracked with client IDs and deduplicated in the Activity Log [[GH-12820](https://github.com/hashicorp/vault/pull/12820)]
```

Author: Updated
